### ISSUE TITLE
Cannot install cheat engine

### ISSUE THREAD

**Issue by zetec-m** [cheat-engine/cheat-engine#2188]
Let's get straight to the point, I'm having the same "Forever loading bar" error everyone else used to have.

First things first, I don't have any virus protection software, and all windows defender settings are off, meaning you do not have to tell me to check #2099 because it does not work, it's the program not responding every time I open the installer.

The weird part is, it wouldn't install the time before I reset my computer, so it's a reoccurring issue for me at least.
I wait at least 10 minutes, then it brings me to an Error, Couldn't Install screen or somethin and I had to cancel out of it.

**Comment by cheat-engine** [comment]
did you check your firewalls and ad blockers ? e.g a pihole will likely block it

**Comment by zetec-m** [comment]
> did you check your firewalls and ad blockers ? e.g a pihole will likely block it

bruh i stated I checked, it's not me everything is off and it's excluded from my firewall and or whatever Is there that should be considered an "Anti Virus" in windows, I'm telling you man it's not me and I've tried multiple installers

**Comment by fade2gray** [comment]
@zetec-m  you mean the "Forever loading bar" as seen [here](https://github.com/cheat-engine/cheat-engine/issues/2096) and [here](https://github.com/cheat-engine/cheat-engine/issues/2141)?

You see, when people don't respond after being given advice, I think it's safe to assume the advice given successfully provided the solution - good luck.

**Comment by zetec-m** [comment]
@fade2gray 

> You see, when people don't respond after being given advice, I think it's safe to assume the advice given successfully provided the solution - good luck.

No, I just simply don't check github because it's not relevant to me, but yes the "Forever loading bar" then after it "loads" the installation box pops up saying Error.

**Comment by fade2gray** [comment]
Download and installation is a two-step process.
1. Download and run the installer.
2. The installer then connects to another site and downloads the rest of the files required to complete installation.

If you are seeing the following error, it means step 2 cannot complete.

![2022-05-17_09-59-15](https://user-images.githubusercontent.com/1332009/168782806-07566f0e-cb92-4f1c-a70f-0f98a15e5f2c.png)

You may need to add an exclusion in Windows Defender for your temp folder `C:\Users\<username>\AppData\Local\Temp`.

Otherwise, the problem could be due to…
- Disconnecting from the internet before running the installer.
- Connecting to the internet via a VPN or Proxy.
- Your ISP blocking the connection to the second site.
- You use a DNS sinkhole like a Pi-Hole for ad blocking etc.
- Having modified firewall settings.

**Comment by zetec-m** [comment]
Huh. The hell, it worked after adding the temp folder as an exclusion. Weird, thanks
